packages/runner/lib/monitor.ts

```diff
@@ -0,0 +1,156 @@
+                        environment_id: environmentId,
+                        activity_log_id: activityLogId,
+                        content: `Memory usage of nango scripts is high: ${memoryUsagePercentage.toFixed(2)}% of the total available memory.`,
+                        timestamp: Date.now()
+                    } as ActivityLogMessage
+                })
+            );
+        }
+    }
+
+    private checkIdle(): NodeJS.Timeout {
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        return setInterval(async () => {
+            if (this.tracked.size == 0) {
+                const idleTimeMs = Date.now() - this.lastIdleTrackingDate;
+                if (idleTimeMs > this.idleMaxDurationMs) {
+                    logger.info(`Runner '${this.runnerId}' idle for more than ${this.idleMaxDurationMs}ms`);
+                    await this.send(
+                        'post',
+                        `/idle`,
+                        superjson.stringify({
+                            runnerId: this.runnerId,
+                            idleTimeMs
+                        })
+                    );
+                    this.lastIdleTrackingDate = Date.now();
+                }
+            }
+        }, 10000);
+    }
+
+    private async send(method: string, endpoint: string, superjsonEncodedBody: string): Promise<void> {
+        if (!this.baseUrl) {
+            return;
+        }
+        const url = `${this.baseUrl}${endpoint}`;
+        try {
+            const res = await fetch(url, {
+                method: method,
+                headers: {
+                    Accept: 'application/json',
+                    'Content-Type': 'application/json'
+                },
+                body: superjsonEncodedBody
+            });
+            if (res.status !== 200) {
+                logger.error(`Error reporting '${superjsonEncodedBody}' to '${url}': ${JSON.stringify(await res.json())}`);
+            }
+        } catch (err) {
+            logger.error(`Error reporting '${superjsonEncodedBody}' to '${url}': ${JSON.stringify(err)}`);
+        }
+    }
+}
+
+function getRenderTotalMemoryInBytes(): number {
+    try {
+        const output = fs.readFileSync('/sys/fs/cgroup/memory.max', 'utf-8');
+        const memoryLimitInBytes = parseInt(output.trim(), 10);
+        return memoryLimitInBytes;
+    } catch (_error) {
+        return 0;
+    }
+}
+
+function getTotalMemoryInBytes(): number {
+    // when running inside a container, os.totalmem() returns the total memory of the system, not the memory limit of the container
+    // see: https://github.com/nodejs/node/issues/51095
+    // process.constrainedMemory() is supposed to return the memory limit of the container but it doesn't work on Render
+    // so we need to use a workaround to get the memory limit of the container on Render
+    return process.constrainedMemory() || getRenderTotalMemoryInBytes() || os.totalmem();
```

Comment:
This is incredibly fragile but I don't have a better solution since `os.totalmem` doesn't return the container memory limit :( Let me know what you think